lib/internal/process/per_thread.js

```diff
@@ -182,8 +182,9 @@ function wrapProcessMethods(binding) {
   function exit(code) {
     process.off('exit', handleProcessExit);
 
-    if (code || code === 0)
+    if (code !== undefined) {
```

Comment:
Maybe it's best to always call the setter as long as any value was supplied?
```suggestion
    if (arguments.length !== 0) {
```